addons/account/static/src/js/bills_tree_upload.js

```diff
@@ -1,37 +1,93 @@
+odoo.define('web.UploadBillMixin', function (require) {
+"use strict";
+
+    var core = require('web.core');
+    var _t = core._t;
+
+    var UploadBillMixin = {
+        _onUpload: function (event) {
```

Comment:
Seems like this mixin is a copy paste from the mechanism of the document module with a few changes.
I did not like the fact that the code was for 80% of it duplicated, so I looked a bit to see if it was possible to refactor it and by looking at how the upload attachment process works, I found a way to do it which seems better as it is how most modules that upload a file works. I wanted to test it, so I happened to write the changes for the javascript (and xml) myself. You can have a look here at this commit to see the changes: https://github.com/odoo-dev/odoo/commit/aeb4df9f26fa75fd32712af38b5b9e4cd99118b6 You can use them and check if everything still works with it.

It feels more readable and less hacky than manually creating a form in javascript and then applying a regex on the result to get the ids of the attachment. We use the original mechanism of upload that once it finished, returns a callback method (if specified) with a list of attachment as parameters. If you've got any questions, don't hesitate to ask.
Also, among the changes, I've removed the upload button from the invoice list view as it seemed like it should not have been there. I've also remove the button_template from the kanban widget as this was doing nothing and was wrong (there is no button in top left corner of accounting kanban view and if we wanted to add one, we should have used another template than the one from the listView as they don't entirely have the same structure).